pkg/analyze/host_system_packages.go

```diff
@@ -0,0 +1,164 @@
+package analyzer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"text/template"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/replicatedhq/troubleshoot/pkg/collect"
+)
+
+type AnalyzeHostSystemPackages struct {
+	hostAnalyzer *troubleshootv1beta2.SystemPackagesAnalyze
+}
+
+func (a *AnalyzeHostSystemPackages) Title() string {
+	return hostAnalyzerTitleOrDefault(a.hostAnalyzer.AnalyzeMeta, "System Packages")
+}
+
+func (a *AnalyzeHostSystemPackages) IsExcluded() (bool, error) {
+	return isExcluded(a.hostAnalyzer.Exclude)
+}
+
+func (a *AnalyzeHostSystemPackages) Analyze(getCollectedFileContents func(string) ([]byte, error)) ([]*AnalyzeResult, error) {
+	hostAnalyzer := a.hostAnalyzer
+
+	contents, err := getCollectedFileContents("system/packages.json")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get collected file")
+	}
+
+	var infos []collect.SystemPackageInfo
+	if err := json.Unmarshal(contents, &infos); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal system packages info")
+	}
+
+	allResults := []*AnalyzeResult{}
+
+	for _, info := range infos {
+		for _, outcome := range hostAnalyzer.Outcomes {
+			isInstalled := isPackageInstalled(info)
+			r := AnalyzeResult{}
+			when := ""
+
+			if outcome.Fail != nil {
+				r.IsFail = true
+				r.Message = outcome.Fail.Message
+				r.URI = outcome.Fail.URI
+				when = outcome.Fail.When
+			} else if outcome.Warn != nil {
+				r.IsWarn = true
+				r.Message = outcome.Warn.Message
+				r.URI = outcome.Warn.URI
+				when = outcome.Warn.When
+			} else if outcome.Pass != nil {
+				r.IsPass = true
+				r.Message = outcome.Pass.Message
+				r.URI = outcome.Pass.URI
+				when = outcome.Pass.When
+			} else {
+				fmt.Println("error: found an empty outcome in a systemPackages analyzer") // don't stop
```

Comment:
probably shouldn't print here

Reply:
it's for debugging purposes, otherwise we'd never know what happened or why there were no results in this case

Reply:
can this print to stderr?